crypto/getenv.c

```diff
@@ -13,9 +13,45 @@
 
 #include <stdlib.h>
 #include "internal/cryptlib.h"
+#include "e_os.h"
 
 char *ossl_safe_getenv(const char *name)
 {
+#if (defined(OPENSSL_SYS_WINDOWS)) && defined(CP_UTF8) && !defined(_WIN32_WCE)
```

Comment:
Maybe it is worth looking at some other files that do this kind of thing internally already e.g ```randfile.c:RAND_file_name()```
The #ifdef looks a bit different.

Reply:
Why _WIN32 is better than OPENSSL_SYS_WINDOWS?

Reply:
This is debatable of course...  `OPENSSL_SYS_` macros were defined back when compilers didn't necessarily have reliable pre-defined platform macros.  Now we're almost two decades later, and the pre-defined macros have become much more reliable.

We have both variants sprinkled throughout the code, and most people prefer to trust the compiler these days (and frankly, that puts less strain on us to maintain the `OPENSSL_SYS_` macros)